src/webgpu/shader/validation/expression/call/builtin/cos.spec.ts

```diff
@@ -79,3 +79,33 @@ Validates that scalar and vector integer arguments are rejected by ${builtin}()
       'constant'
     );
   });
+
+const kGoodArgs = '(1.1)';
+const kBadArgs = {
+  no_parens: '',
+  // Bad number of args
+  '0args': '()',
+  '2args': '(1.0,2.0)',
+  // Bad value for arg 0
+  '0i32': '(1i)',
+  '0u32': '(1u)',
+  '0bool': '(false)',
+  '0vec2u': '(vec2u())',
+  '0array': '(array(1.1,2.2))',
+  '0struct': '(modf(2.2))',
+};
```

Comment:
I find the number prefix a bit confusing here - suggested renames.

I also suggest folding `kGoodArgs` and `kBadArgs`, then changing `t.expectCompileResult(t.params.arg == 'pass', ...)`, so you get a single positive case.

```suggestion
const kArgCases = {
  pass: '(1.1)',
  fail_no_parens: '',
  fail_no_args: '()',
  fail_two_args: '(1.0,2.0)',
  fail_1i: '(1i)',
  fail_1u: '(1u)',
  fail_false: '(false)',
  fail_vec2u: '(vec2u())',
  fail_array: '(array(1.1,2.2))',
  fail_struct: '(modf(2.2))',
};
```

Reply:
I'd rather use "good" and "bad" because "pass" and "fail" are judgements about tests and sometimes the test passes when the args are bad.